src/main/java/frc/robot/subsystems/ShooterSubsystem.java

```diff
@@ -157,224 +172,137 @@ public void updatePID() {
-        currentRPM = Constants.Shooter.TARMACRPM;
+      case TARMAC: // aimMode used to shoot into the high goal from ~2ft
+        adjustHood(aimMode.getAngle());
+        windFlywheel(aimMode.getRPM());
         break;
-      case 4: // Case for TEST mode, just takes an RPM and winds
-        windFlywheel(DShooterRPMInput.getDouble(0));
-        currentRPM = DShooterRPMInput.getDouble(0);
+      case LAUNCH: // aimMode used to shoot into the high goal from the launchpad
+        adjustHood(aimMode.getAngle());
+        windFlywheel(aimMode.getRPM());
         break;
-      case 5:
+      case AUTO: // aimMode used to automatically shoot into the high goal
         windFlywheel(lookup(getDistance())[0]);
-        currentRPM = lookup(getDistance())[0];
-
+        //TODO: Also adjust hood here
+        break;
+      case TEST: // aimMode to take a RPM from the dashboard
+        windFlywheel(DShooterRPMInput.getDouble(0));
         break;
     }
   }
 
+  public void updateSmartDashboard() {
+    SmartDashboard.putString("AimMode", "");
+    SmartDashboard.putNumber("IAccum",flywheelPID.getIAccum());
+    SmartDashboard.putNumber("Distance", getDistance());
+    SmartDashboard.putNumber("RPM", flywheelEncoder.getVelocity());
+  }
+
   @Override
   public void periodic() {
     // This method will be called once per scheduler run
-    // SmartDashboard.putNumber("IAccum",KShooterController.getIAccum());
-    // SmartDashboard.putNumber("dist", getDistance());
-    if (DShooterRPM.getDouble(0.0) != shooter_motorController.getEncoder().getVelocity()) {
-      DShooterRPM.setDouble(shooter_motorController.getEncoder().getVelocity());
+    if (DShooterRPM.getDouble(0.0) != flywheelEncoder.getVelocity()) {
+      DShooterRPM.setDouble(flywheelEncoder.getVelocity());
     }
     if (DDistance.getDouble(0.0) != getDistance()) {
       DDistance.setDouble(getDistance());
     }
     if (dashTunePid.getBoolean(false)) {
-      if ((shooter_motorController.getPID().getP() != PID_P.getDouble(0))
-          || (shooter_motorController.getPID().getI() != PID_I.getDouble(0))
-          || (shooter_motorController.getPID().getD() != PID_D.getDouble(0))) {
+      if ((flywheelPID.getP() != PID_P.getDouble(0))
+          || (flywheelPID.getI() != PID_I.getDouble(0))
+          || (flywheelPID.getD() != PID_D.getDouble(0))) {
         updatePID();
       }
-
       // dashTunePid.setBoolean(false);
-
     }
-    if (ShooterReverted.getBoolean(false) != shooter_motorController.getEncoder().getInverted()) {
-      shooter_motorController.setInverted(ShooterReverted.getBoolean(false));
+    if (ShooterReverted.getBoolean(false) != flywheelEncoder.getInverted()) {
+      flywheelController.setInverted(ShooterReverted.getBoolean(false));
     }
-    if (DShootingMode.getDouble(0) != aimMode) {
-      aimMode = DShootingMode.getDouble(0);
+    if (DShootingMode.getDouble(0) != aimMode.ordinal()) {
+      setAimMode((int)DShootingMode.getDouble(0));
```

Comment:
enums have a .toString() method that could make the Shuffleboard entry easier to parse by the operator. Makes setting the aimMode from shuffleboard a bit trickier though.

Reply:
Yeah, my primary problem with this was that it makes setting the aimMode hard, ideally we would be setting the enum, but I don't know how to do that with shuffleboard. I think we opened issue #102 for this.